### PR TITLE
Feat remove public ids

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: src/keys/keypair.rs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+use crate::{Error, Result};
+use crate::{PublicKey, SecretKey, Signature, SignatureShare};
+
+use ed25519_dalek::Signer;
+use rand::{CryptoRng, Rng};
+use serde::{Deserialize, Serialize};
+use std::fmt::{self, Debug, Formatter};
+use threshold_crypto::{self, serde_impl::SerdeSecret};
+use unwrap::unwrap;
```
Comment (dirvine) [inline src/keys/keypair.rs]: We still need to remove this.

File: src/keys/keypair.rs [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    /// Ed25519 keypair.
+    Ed25519(ed25519_dalek::Keypair),
+    /// BLS keypair.
+    Bls(BlsKeypair),
+    /// BLS keypair share.
+    BlsShare(BlsKeypairShare),
+}
+
+// Need to manually implement this due to a missing impl in `Ed25519::Keypair`.
+impl Clone for Keypair {
```
Comment (dirvine) [inline src/keys/keypair.rs]: We likely don't need Clone and the unwrap if we use Arc (for now anyway)

File: src/keys/keypair.rs [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        ]
+    }
+
+    fn gen_keys() -> Vec<PublicKey> {
+        gen_keypairs().iter().map(PublicKey::from).collect()
+    }
+
+    // Test serialising and deserialising key pairs.
+    #[test]
+    fn serialisation_key_pair() {
```
Comment (dirvine) [inline src/keys/keypair.rs]: if we make this a `Result<>` we can use a `?` and not unwrap (we need to kill that crate anyway. If we need unwrap we can use the std `.unwrap("comment")`

File: src/keys/keypair.rs [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+    /// Signs with the underlying keypair.
+    pub fn sign(&self, data: &[u8]) -> Signature {
+        match self {
+            Self::Ed25519(keypair) => Signature::Ed25519(keypair.sign(&data)),
+            Self::Bls(keypair) => Signature::Bls(keypair.secret.sign(data)),
+            Self::BlsShare(keypair) => {
+                let index = keypair.index;
+                let share = keypair.secret.sign(data);
+                Signature::BlsShare(SignatureShare { index, share })
```
Comment (Yoga07) [inline src/keys/keypair.rs]:
```suggestion
                Signature::BlsShare(SignatureShare { index: keypair.index, share: keypair.secret.sign(data) })
```
This way we can remove the unnecessary init of two vars above. We don't seem to use them elsewhere.

File: src/keys/keypair.rs [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+            public,
+        };
+        Self::Bls(keypair)
+    }
+}
+
+impl From<SerdeSecret<threshold_crypto::SecretKey>> for Keypair {
+    fn from(sk: SerdeSecret<threshold_crypto::SecretKey>) -> Self {
+        let public = sk.public_key();
+        let keypair = BlsKeypair { secret: sk, public };
```
Comment (Yoga07) [inline src/keys/keypair.rs]: Same as above. Don't seem to use `public` and `keypair` elsewhere.

File: src/keys/keypair.rs [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+        Self::Bls(keypair)
+    }
+}
+
+impl From<&SerdeSecret<threshold_crypto::SecretKey>> for Keypair {
+    fn from(sk: &SerdeSecret<threshold_crypto::SecretKey>) -> Self {
+        let public = sk.public_key();
```
Comment (Yoga07) [inline src/keys/keypair.rs]: Same as above.

**Review by Yoga07 — DISMISSED** [review]
Suggested a few code style changes. LGTM otherwise :+1:

**Review by Yoga07 — APPROVED** [review]

**Pull request merged**